### ISSUE TITLE
Terraform deploy grafana dashboard: Error: rpc error: code = Unavailable desc = transport is closing

### ISSUE THREAD

**Issue by MinsonTFW** [grafana/terraform-provider-grafana#238]
### Terraform Version
* Terraform: 
   - Terraform v0.14.11
   - Terraform v0.15.5
   - provider registry.terraform.io/grafana/grafana v1.13.0
   - provider registry.terraform.io/hashicorp/http v2.1.0
   - provider registry.terraform.io/hashicorp/template v2.2.0
* Terraform Grafana Provider: 1.11.0, 1.12.0, 1.13.0, 1.13.1(later than 1.10.0)
* Grafana: v6.7.1 (ca6d08d5cb)

### Affected Resource(s)
* grafana_dashboard

### Terraform Configuration Files
```hcl
terraform {
  required_providers {
    grafana = {
      source  = "grafana/grafana"
    }
  }
}

provider "grafana" {
  url  = "xxx"
  auth = "xxx"
}

resource "grafana_alert_notification" "email_alert" {
  count      = 1
  name       = "xxx"
  type       = "email"
  is_default = false

  settings = {
    "addresses"   = "xxx@xxx.com"
    "uploadImage" = "true"
  }
}

data "http" "uid" {
  url = format("%sapi/alert-notifications/%s", "xxx", join(", ", grafana_alert_notification.email_alert[*].id))
  request_headers = {
    Authorization = format("Bearer %s", "xxx")
  }

  depends_on = [
    grafana_alert_notification.email_alert,
  ]

}

data "template_file" "dash_tpl" {
  count    = 1
  template = file("dash.tpl")
  vars = {
    region           = "eu-central-1"
    loadbalancername = "xxx"
    loadbalancerarn  = "xxx"
    topic            = "xxx"
    stage            = "xxx"
    datasource       = "Toolbox"
    uid              = jsondecode(data.http.uid.body)["uid"]
  }
}

resource "grafana_dashboard" "metrics" {
  count       = 1
  config_json = data.template_file.dash_tpl[count.index].rendered
  lifecycle {
    ignore_changes = [
      config_json
    ]
  }
}
```

template file: https://gist.github.com/MinsonTFW/887410deee3fcd825eaa8ad5492fe116

### Debug Output
For Terraform v0.14.11: https://gist.github.com/MinsonTFW/6298e6a9a741020e917e6af8a2b25003
For Terraform v0.15.5: https://gist.github.com/MinsonTFW/57b43bba71bea6569df8b4088a5f375d
For Terraform v1.0.2 with grafana provider v1.13.1: https://gist.github.com/MinsonTFW/6db58e81fff10f68a38a854b4406e83c

### Panic Output
For Terraform v0.14.11: https://gist.github.com/MinsonTFW/dc842371c7bffb5a78bc511816d4ad2b
For Terraform v0.15.5: no crash_log generate

### Expected Behavior
there should be no error message while plan to deploy grafana resource via Terraform

### Actual Behavior
While plan to deploy grafana dashboard via Terraform, got the Error: rpc error: code = Unavailable desc = transport is closing

### Steps to Reproduce
1. `terraform init`
2. `terraform plan`

### Workaround:
I can successful get the output with terraform plan if I stick the grafana provider version to: version = "~> 1.10.0"

Would real appreciated if anyone can help and have a check, thanks!

**Comment by ninjatec** [comment]
Further to this workaround above, I have found that once you have performed a "terraform apply" the workaround can be removed.

**Comment by akshaychopra5207** [comment]
HI @flokli

I am still getting the same error after I have upgraded the version to 1.13.3

Versions I am using
Terraform v0.13.4

provider digitalplatform.local/maf/digitalplatform v1.0.1
provider registry.terraform.io/grafana/grafana v1.13.3
provider registry.terraform.io/hashicorp/azurerm v2.43.0
provider registry.terraform.io/hashicorp/http v1.2.0

Error during plan
```
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: panic: Attempted to unmarshal invalid JSON. This unexpectedly got past schema validation.
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: 
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: goroutine 40 [running]:
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: github.com/grafana/terraform-provider-grafana/grafana.unmarshalDashboardConfigJSON(0x19ba146, 0x24, 0x18747e0)
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         github.com/grafana/terraform-provider-grafana/grafana/resource_dashboard.go:268 +0x106
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: github.com/grafana/terraform-provider-grafana/grafana.normalizeDashboardConfigJSON(0x18747e0, 0xc0000aebd0, 0xb, 0x18747e0)
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         github.com/grafana/terraform-provider-grafana/grafana/resource_dashboard.go:299 +0x145
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diffString(0xc0000aa870, 0x19a4dad, 0xb, 0xc0003b6280, 0xc000375508, 0x1ab41d8, 0xc000175b80, 0x203000, 0x0, 0x0)
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/schema.go:1375 +0x6e3
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.diff(0xc0000aa870, 0x19a4dad, 0xb, 0xc0003b6280, 0xc00089a780, 0x1ab41d8, 0xc000175b80, 0x0, 0x0, 0x0)
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/schema.go:957 +0x4e8
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.schemaMap.Diff(0xc0000aa870, 0x1ab05b8, 0xc0002e42c0, 0xc0001b2bd0, 0xc000302d80, 0x0, 0x184bce0, 0xc0002c6dc0, 0x1f8ff00, 0x18747e0, ...)
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/schema.go:522 +0x219
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).SimpleDiff(0xc0003b4380, 0x1ab05b8, 0xc0002e42c0, 0xc0001b2bd0, 0xc000302d80, 0x184bce0, 0xc0002c6dc0, 0x0, 0x0, 0x0)
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:506 +0xa5
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).PlanResourceChange(0xc0000da078, 0x1ab05b8, 0xc0002e42c0, 0xc0002d6b90, 0xc0002e42c0, 0x1963d60, 0xc0002dab00)
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:693 +0x7c5
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).PlanResourceChange(0xc00082c580, 0x1ab0660, 0xc0002e42c0, 0xc0001b29a0, 0xc00082c580, 0xc0002dab40, 0xc000525ba0)
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:315 +0xb5
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_PlanResourceChange_Handler(0x1963d60, 0xc00082c580, 0x1ab0660, 0xc0002dab40, 0xc0000a8840, 0x0, 0x1ab0660, 0xc0002dab40, 0xc000526000, 0xa3b4)
2021-08-04T13:15:35.979+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:362 +0x214
2021-08-04T13:15:35.980+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00027c8c0, 0x1ab7f58, 0xc000135500, 0xc00022c360, 0xc000302690, 0x1f51088, 0x0, 0x0, 0x0)
2021-08-04T13:15:35.980+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         google.golang.org/grpc@v1.38.0/server.go:1286 +0x52b
2021-08-04T13:15:35.980+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: google.golang.org/grpc.(*Server).handleStream(0xc00027c8c0, 0x1ab7f58, 0xc000135500, 0xc00022c360, 0x0)
2021-08-04T13:15:35.980+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         google.golang.org/grpc@v1.38.0/server.go:1609 +0xd0c
2021-08-04T13:15:35.980+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000b6210, 0xc00027c8c0, 0x1ab7f58, 0xc000135500, 0xc00022c360)
2021-08-04T13:15:35.980+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         google.golang.org/grpc@v1.38.0/server.go:934 +0xab
2021-08-04T13:15:35.980+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-08-04T13:15:35.980+0400 [DEBUG] plugin.terraform-provider-grafana_v1.13.3:         google.golang.org/grpc@v1.38.0/server.go:932 +0x1fd
2021-08-04T13:15:35.981+0400 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-08-04T13:15:35.982+0400 [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/grafana/grafana/1.13.3/darwin_amd64/terraform-provider-grafana_v1.13.3 pid=90788 error="exit status 2"
2021/08/04 13:15:35 [ERROR] eval: *terraform.EvalDiff, err: rpc error: code = Unavailable desc = transport is closing
2021/08/04 13:15:35 [ERROR] eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
```

**Comment by ryanhestin** [comment]
I just ran into this issue too while processing updates from Terraform 0.12.29 to 0.12.31 across our codebase, which enabled us to catch up on providers. In rare cases (for reasons beyond me at this point), two of the states (across a few hundred total) have ended up in a broken Grafana provider state on plans.

Having previously been on 1.9.0 of the plugin, I pinned it there and was immediately successful in planning. I then stepped up minor versions until I hit a version that caused the break. In my case, I made it through 1.11.0 fine. Once I got to 1.12.0, I encountered the error. `TRACE` logs don't seem to indicate anything. We were even thrown for a loop when the last thing in the log before the RPC failure line was a bunch of AWS provider bits D:

I don't have much more to offer at this point. I think we'll pin to 1.9.0 across our internal modules and kick the can waiting for a fix.

Edit: We ended up pinning at the implementing repo level since about half the repos were already shoved up to the 1.14 provider.
